scheduler/pkg/agent/repository/triton/triton_test.go

```diff
@@ -336,3 +338,85 @@ func TestCopyNonConfigFilesToModelRepo(t *testing.T) {
 		})
 	}
 }
+
+func TestGetModelConfig(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	type test struct {
+		name                  string
+		modelConfig           *pb.ModelConfig
+		expectedInstanceCount uint32
+	}
+
+	tests := []test{
+		{
+			name: "onnxruntime backend without a count defaults to 2",
+			modelConfig: &pb.ModelConfig{
+				Backend:       "onnxruntime",
+				InstanceGroup: []*pb.ModelInstanceGroup{{Kind: pb.ModelInstanceGroup_KIND_CPU}},
+			},
+			expectedInstanceCount: 2,
+		}, {
+			name: "tensorflow backend without a count defaults to 2",
+			modelConfig: &pb.ModelConfig{
+				Backend:       "tensorflow",
+				InstanceGroup: []*pb.ModelInstanceGroup{{Kind: pb.ModelInstanceGroup_KIND_CPU}},
+			},
+			expectedInstanceCount: 2,
+		}, {
+			name: "other backend without a count defaults to 2",
+			modelConfig: &pb.ModelConfig{
+				Backend:       "other",
+				InstanceGroup: []*pb.ModelInstanceGroup{{Kind: pb.ModelInstanceGroup_KIND_CPU}},
+			},
+			expectedInstanceCount: 1,
+		}, {
+			name: "not KIND_CPU defaults to 1",
```

Comment:
Question: This is 1 specifically because atm we only expose system RAM as something a model can request as required memory, right? I'm asking because by default Triton creates one model copy per available GPU.

Reply:
Defaulting to one, as the number of GPUs available is unknown for now.